### PR TITLE
Plant Routes

### PR DESCRIPTION
**Pull request by spencer-rafada**
- I merged main to my branch and run the formatting and linter so there will be no merge conflicts now.
- Added Plant routes and refactored it already

### CODE REVIEW

**Review by jjossie — APPROVED** [review]
Love it, high quality stuff Spencer 🤝

File: src/routes/plant.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+          .send({ message: `Plant with ID: ${req.params.plantId}` });
+      }
+    } catch (error) {
+      console.log(error);
+    }
+  }
+);
+
+// Are we going to add a potential archive list?
+plantRouter.delete('/:plantId', requiresAuth());
```
Comment (Jacksonckc) [inline src/routes/plant.ts:74]: is this supposed to be empty?

File: src/routes/plant.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    else res.status(200).send(result);
+  } catch (error) {
+    console.log(error);
+  }
+});
+
+plantRouter.post('/', requiresAuth(), async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const newPlant: HydratedDocument<IPlant> = { ...req.body };
+    const result = await addPlant(newPlant);
```
Comment (Jacksonckc) [inline src/routes/plant.ts:33]: Are we going to check for the userLevel before we call this function? Or did you do it somewhere else already.

File: src/routes/plant.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  async (req: Request, res: Response, next: NextFunction) => {
+    try {
+      const plant: IPlant = {
+        ...req.body
+      };
+      // this is filter to look plantId
+      const filter = { _id: req.params.plantId };
+
+      // TODO: Refactor this to plant controller
+      const result = await updatePlant(filter, plant);
```
Comment (Jacksonckc) [inline src/routes/plant.ts:58]: Same question as above

**Review by Jacksonckc — CHANGES_REQUESTED** [review]
Please take a look at my questions thank you

File: src/routes/plant.ts (hunk 1 above)
Comment (spencer-rafada) [reply src/routes/plant.ts]: I've worked on the functionality for this route.

File: src/routes/plant.ts (hunk 3 above)
Comment (spencer-rafada) [reply src/routes/plant.ts]: I just finished adding the check for userLevel for PUT, POST, and DELETE route.

**Pull request merged**